src/script/common/c_converter.h

```diff
@@ -42,31 +42,51 @@ bool               getboolfield_default(lua_State *L, int table,
                              const char *fieldname, bool default_);
 float              getfloatfield_default(lua_State *L, int table,
                              const char *fieldname, float default_);
-int                getintfield_default           (lua_State *L, int table,
+int                getintfield_default(lua_State *L, int table,
                              const char *fieldname, int default_);
 
+template<typename T>
+bool getintfield(lua_State *L, int table,
+		const char *fieldname, T &result)
+{
+	lua_getfield(L, table, fieldname);
+	bool got = false;
+	if (lua_isnumber(L, -1)){
+		result = lua_tointeger(L, -1);
+		got = true;
+	}
+	lua_pop(L, 1);
+	return got;
+}
+
+template<class T>
+bool getv3field(lua_State *L, int index,
```

Comment:
Maybe the name should make it clear it is for v3 ints? `getv3intField`?

Reply:
Right. Changed that one.